syntax = "proto3";

import "get_plaform_details.proto";
import "coils.proto";
import "fade_light.proto";
import "switch.proto";

service HardwarePlatform {
  rpc GetPlatformDetails(GetPlatformDetailsRequest) returns (GetPlatformDetailsResponse);
  rpc GetSwitchChanges(SwitchChangesRequest) returns (stream SwitchChanges);
  rpc LightFade(FadeLightRequest) returns (FadeLightResponse);
  rpc CoilPulse(PulseCoilRequest) returns (CoilResponse);
  rpc CoilEnable(EnableCoilRequest) returns (CoilResponse);
  rpc CoilDisable(DisableCoilRequest) returns (CoilResponse);
  rpc ConfigureHardwareRule(ConfigureHardwareRuleRequest) returns (CoilResponse);
  rpc RemoveHardwareRule(RemoveHardwareRuleRequest) returns (CoilResponse);
}
